python/lsst/ctrl/bps/wms_service.py

```diff
@@ -82,6 +84,97 @@ class WmsStates(Enum):
     """At least one of the parents failed or can't be run."""
 
 
+class WmsSpecificInfo:
+    """Class representing WMS specific information.
+
+    Each piece of information is split into two parts: a template and
+    a context. The template is a string that can contain literal text and/or
+    *named* replacement fields delimited by braces ``{}``. The context is
+    a mapping between the variables that names correspond to the replacement
+    fields in the template and their values.
+
+    To produce a human-readable representation of the information, e.g., for
+    logging purposes, it needs to be rendered first to combine these two parts.
+    On the other hand, the context alone might be sufficient if the provided
+    information is being ingested to a database.
+    """
+
+    def __init__(self):
+        self._context: dict[str, Any] = {}
+        self._templates: list[str] = []
+
+    def __bool__(self):
+        return bool(self._templates)
+
+    @property
+    def context(self) -> dict[str, Any]:
+        """The context that will be used to render the information.
+
+        Returns
+        -------
+        context : `dict` [`str`, `Any`]
+            The dictionary representing the mapping between *every* template
+            variable and its value.
+        """
+        return self._context
```

Comment:
No, it the user should not modify `_context` (and `_template` for that matter) outside of the class as it may prevent the instance of the `WmsSpecificInfo` from rendering in print/log messages. I will modify both properties to return copies of the respective dictionaries.